src/lib/rest/StringFilter.cpp

```diff
@@ -0,0 +1,1101 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```

Comment:
This class now encapsulated all the complexity related with filtering. Good!

However, now old qStringFilters() and children functions in mongoBackend should disapear in favour of this new class. Otherwise, we would have duplicated code for the same functionality (with the aggravating circumstance that this is a particularly complex piece of code).